ui/appui-react/src/appui-react/statusfields/message-center/MessageCenterField.tsx

```diff
@@ -0,0 +1,169 @@
+  const [messages, setMessages] = React.useState(MessageManager.messages);
+  const [notify, setNotify] = React.useState(false);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [status, setStatus] =
+    React.useState<NotificationMarkerStatus>("primary");
+
+  const indicatorRef = React.useRef<React.RefObject>();
+  const title = UiFramework.translate("messageCenter.messages");
+
+  const handleOpenChange = (isOpenState: boolean) => {
+    setNotify(false);
+    setIsOpen(isOpenState);
+  };
+
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
+    );
+  };
+
+  const determineStatus = () => {
+    const message = [...MessageManager.messages].pop();
+
+    if (message) {
+      if (isProblemStatus(message)) {
+        return setStatus("negative");
+      } else if (message.priority === OutputMessagePriority.Success) {
+        return setStatus("positive");
+      } else {
+        return setStatus("primary");
+      }
+    }
+    return;
+  };
+
+  React.useEffect(() => {
+    return MessageManager.onOpenMessageCenterEvent.addListener(() => {
+      handleOpenChange(true);
+    });
+  });
+
+  React.useEffect(() => {
+    MessageManager.registerAnimateOutToElement(indicatorRef.current);
+
+    return MessageManager.onMessagesUpdatedEvent.addListener(() => {
+      MessageManager.messages.length > 0 ? setNotify(true) : setNotify(false);
+      setMessages(MessageManager.messages);
+      determineStatus();
+    });
+  });
+
+  const getMessages = (tab: string) => {
+    return [...messages].reverse().map((message, index) => {
+      if (messages.length > 0) {
+        const iconClassName = MessageManager.getIconClassName(message);
+        const iconSpec = MessageManager.getIconSpecFromDetails(message);
+
+        if ((tab === "error" && isProblemStatus(message)) || tab === "all") {
+          return (
+            <MessageCenterMessage
+              key={index.toString()}
+              message={message.briefMessage}
+              details={message.detailedMessage}
+              icon={<Icon iconSpec={iconSpec} className={iconClassName} />}
+            />
+          );
+        }
+        return;
+      } else {
+        return (
+          <span className="uifw-message-prompt" key={`${index.toString()}`}>
+            No Messages.
+          </span>
+        );
+      }
+    });
+  };
+
+  const tabs = (
+    <>
+      {["all", "error"].map((tabType) => {
+        return (
+          <Tabs.Panel value={tabType} key={tabType}>
+            <div className="uifw-statusFields-messageCenter-messageCenterField">
+              {getMessages(tabType)}
+            </div>
+          </Tabs.Panel>
+        );
+      })}
+    </>
+  );
+
+  return (
+    <Popover
+      visible={isOpen}
```

Comment:
When using `visible`, you need to also use `onVisibleChange`, to avoid throwing away the built-in Popover triggers.

```suggestion
      visible={isOpen}
      onVisibleChange={setIsOpen}
```